packages/remix-server-runtime/links.ts

```diff
@@ -1,13 +1,14 @@
-/**
- * Represents a `<link>` element.
- *
- * WHATWG Specification: https://html.spec.whatwg.org/multipage/semantics.html#the-link-element
- */
-export interface HtmlLinkDescriptor {
+type Primitive = null | undefined | string | number | boolean | symbol | bigint;
+
+type LiteralUnion<LiteralType, BaseType extends Primitive> =
+  | LiteralType
+  | (BaseType & Record<never, never>);
```

Comment:
Is there a reason why we're not using `type-fest` here? 🤔

Reply:
at the time the only type we used from type-fest was this, but i think we actually have the dep now anyways

https://github.com/remix-run/remix/pull/1123#issuecomment-1035921934

Reply:
Just looked into this & we have `type-fest` in `@remix-run/dev`, but it is in `devDependencies` though.

For `@remix-run/server-runtime` it should be in `dependencies` as we're exporting the types.

Reply:
May as well just inline it. No reason to worry about carrying over the dependency for this IMO.